mitodl/micromasters
# Added models for Enrollment and Certificate and code to populate them

#### What are the relevant tickets?
Fixes #677
Rebased on #706

#### What's this PR do?
Adds four models: Certificate, CertificateAccess, Enrollment, EnrollmentAccess. They cache data coming back from edX. After six hours for certificates or 5 minutes for enrollments, the cache is cleared for that user and new data is retrieved.

#### Where should the reviewer start?
dashboard/api.py

#### How should this be manually tested?
Look at your devstack console. Access the dashboard. You should see some requests to the enrollments and possibly certificates APIs. Immediately afterwards access the dashboard again. You should see the same data in the dashboard but no requests to the edX APIs in the console.
